blocks/basic/test/qa_DataSink.cpp

```diff
@@ -243,6 +243,26 @@ bool spinUntil(std::chrono::milliseconds timeout, auto fnc) {
     return false;
 }
 
+template<typename T>
```

Comment:
There are already two existing functions in `DataSetHelper.hpp` that do exactly that (and more):
[gr::dataset::checkDataSetConsistency(..)](https://github.com/fair-acc/gnuradio4/blob/94083cf1f7a2f4fe49384a9efc02e2387ca0f9a2/algorithm/include/gnuradio-4.0/algorithm/dataset/DataSetHelper.hpp#L43C46-L83) and [gr::dataset::verify](https://github.com/fair-acc/gnuradio4/blob/94083cf1f7a2f4fe49384a9efc02e2387ca0f9a2/algorithm/include/gnuradio-4.0/algorithm/dataset/DataSetHelper.hpp#L225-L290).

It would be better to re-use one of these (and perhaps even merge them into one).